src/main/java/io/hops/util/featurestore/FeaturestoreHelper.java

```diff
@@ -1440,29 +1440,34 @@ public static Boolean validatePrimaryKey(Dataset<Row> featuregroupDf, List<Strin
    * @param dtypes the schema of the provided spark dataframe
    * @param description the description about the feature group/training dataset
    */
-  public static void validateMetadata(String name, Tuple2<String, String>[] dtypes, String description) {
-    Pattern namePattern = Pattern.compile("^[a-zA-Z0-9_]+$");
-    if (name.length() > 256 || name.equals("") || !namePattern.matcher(name).matches())
+  public static void validateMetadata(String name, Tuple2<String, String>[] dtypes, String description,
+    FeaturestoreClientSettingsDTO featurestoreClientSettingsDTO) {
+    Pattern namePattern = Pattern.compile(featurestoreClientSettingsDTO.getFeaturestoreRegex());
```

Comment:
I'd not compile the regex every time you call the method.

Reply:
Moved the compilation to class level of FeaturestoreHelper.